src/Controller/Component/AuthorizationComponent.php

```diff
@@ -91,13 +97,32 @@ public function authorize($resource, $action = null)
      */
     public function can($resource, $action = null)
     {
-        $request = $this->getController()->request;
         if ($action === null) {
+            $request = $this->getController()->request;
             $action = $this->getDefaultAction($request);
         }
 
+        try {
+            return $this->_can($resource, $action);
+        } catch (ForbiddenException $ex) {
```

Comment:
Why would `can` even throw one of these exceptions? `can` should return only `bool`, this isn;t supposed to throw any authorization related exceptions.

Reply:
@markstory, we've had a little conversation over there already. As yet unresolved.